packages/widgets/src/Pagination/PageSelect.js

```diff
@@ -0,0 +1,52 @@
+import React from 'react'
+import propTypes from '@dhis2/prop-types'
+import { SingleSelect, SingleSelectOption } from '@dhis2/ui-core'
+import { spacers } from '@dhis2/ui-constants'
+
+import translate from '../translate'
+
+const createAvailablePages = length =>
+    Array.from({ length }, (_x, i) => (i + 1).toString())
+
+const PageSelect = ({
+    dataTest,
+    pageSelectText,
+    onChange,
+    page,
+    pageCount,
+}) => (
+    <div data-test={`${dataTest}-gotopage`}>
+        <SingleSelect
+            dense
+            selected={page.toString()}
+            onChange={({ selected }) => onChange(parseInt(selected, 10))}
+            className="select"
+            dataTest={`${dataTest}-page-select`}
+            prefix={translate(pageSelectText)}
+        >
+            {createAvailablePages(pageCount).map(availablePage => (
```

Comment:
I think `availablePage` is ambiguous and would prefer `pageNum` or sth like that. But it's  not important at all, so take it or leave it

Reply:
The problem is that these are string representations of numbers, so introducing number into the variable names makes this confusing in another way...